src/middleware/otpauthMiddleware.ts

```diff
@@ -0,0 +1,64 @@
+import randomstring from 'randomstring';
+import Jwt, { VerifyErrors } from 'jsonwebtoken';
+import { Request, Response, NextFunction } from 'express';
+import { config } from 'dotenv';
+import { sendEmail } from '../utils/email';
+import { DataInfo } from '../controllers/OTPcontroller';
+import { decodeToken, verifyToken } from '../utils/tokenGenerate';
+
+
+config();
+
+
+export const generateOtpToken = async (id: string, email:string) => {
+  const code = await randomstring.generate({
+    length: 6,
+    charset: 'numeric',
+  });
+  const mailOptions = {
+    to: email,
+    subject: 'OTP Email verification',
+    template: 'otp',
+    context: {
+      otp: `${code}`,
+    },
+  };
+  await sendEmail(mailOptions);
+  const body = {
+    id,
+    email,
+    otp: code,
+  };
+  const secret = process.env.APP_SECRET;
+  const token = await Jwt.sign({ body }, secret as string, {
+    expiresIn: '180s',
+  });
+  return token;
+};
+
+export const isvalid = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const { token } = req.params;
+    const { otp } = req.body;
+    if (!otp) {
+      return res.status(404).json({ message: 'Invalid OTP' });
+    }
+    
+    const finalresult = async (err: Error, decoded: any) => {
+      const info = decoded as DataInfo;
+     
+      if (info?.body?.otp !== otp) {
+        return res.status(401).json({ message: 'Wrong OTP entered' });
+      }
+      req.user = info;
+      next();
+    };
```

Comment:
why assigning this user again??

Reply:
> you need to verify token only

i do have verify token function in utils and then i called it in isvalid to make sure token is correct

Reply:
the token is valid through isAuthMiddleware

Reply:
this is checking if the otp entered by user is the same one which is sent to their Email